### PR TITLE
Vscode extension for irita wasm

### PR DESCRIPTION
**Pull request by chengwenxi**
Visual Studio Code Extension for IRITA WASM

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,81 @@
...
+
+## Release Notes
+
+See [CHANGELOG.md](CHANGELOG.md)
+
+## Extension Settings
+
+This extension contributes the following settings:
+
+- `Algorithm`: algorithm: sm2 for IRITA, secp256k1 for IRIS
```
Comment (chengwenxi) [inline README.md]:
```suggestion
- `Algorithm`: algorithm: SM2, secp256k1
```

File: README.md [hunk 2]
```diff
@@ -1,2 +1,81 @@
...
+- `ChainID`: chain ID
+- `Denom`: denom
+- `GasLimit`: gas limit
+- `RpcAddress`: RPC address in the form of http://host:port.
```
Comment (chengwenxi) [inline README.md:40]: RPCAddress

File: README.md (hunk 1 above)
Comment (chengwenxi) [reply README.md]: Only used for IRITA, not for IRISHub.

File: extension.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    generate(uri)
+  );
+
+  const compileCmd = vscode.commands.registerCommand('vscode-ext-irita-wasm.compile', async (uri: vscode.Uri | undefined) =>
+    compile(uri)
+  );
+
+  const deployCmd = vscode.commands.registerCommand('vscode-ext-irita-wasm.deploy', async (uri: vscode.Uri | undefined) =>
+    deploy(uri)
+  );
```
Comment (dreamer-zq) [inline extension.ts]: 建议命令如下命名：
  vscode-ext-irita-wasm.generate => irita: Generate contract template
  vscode-ext-irita-wasm.compile => irita: Compile contract
  vscode-ext-irita-wasm.deploy => irita: Deploy contract

File: src/commands/compileCmd.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  }
+
+  if (!fs.statSync(fileName).isDirectory()) {
+    fileName = path.join(fileName, "../");
+  }
+
+  vscode.window.showInformationMessage(`Compiling ${fileNameFromPath(fileName)}...`);
+
+  try {
+    shell.execSync("cd " + fileName + " && cargo wasm");
```
Comment (dreamer-zq) [inline src/commands/compileCmd.ts]: cargo wasm => RUSTFLAGS='-C link-arg=-s' cargo wasm

File: package.json [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+		"eslint": "^7.15.0",
+		"@typescript-eslint/eslint-plugin": "^4.9.0",
+		"@typescript-eslint/parser": "^4.9.0",
+		"glob": "^7.1.6",
+		"mocha": "^8.1.3",
+		"typescript": "^4.1.2",
+		"vscode-test": "^1.4.1"
+	},
+	"dependencies": {
+		"@irisnet/irishub-sdk": "git://github.com/irisnet/irishub-sdk-js.git#bsn/202012",
```
Comment (chengwenxi) [inline package.json]: use https://github.com/bianjieai/irita-sdk-js

**Pull request merged**